zerver/templatetags/app_filters.py

```diff
@@ -167,6 +168,12 @@ def render_markdown_path(
         else:
             raise e
 
+    page_url = context.get("OPEN_GRAPH_URL", "")
+    if markdown_string[0] != "#" and "/api" in page_url:
+        endpoint_name = context["OPEN_GRAPH_URL"].split("/api/")[1].replace("-", "_")
```

Comment:
I think you probably want to be consistent about whether `/api/` is required here.

Also this logic (and others) crash when you visit `/api/invalid`.  So I think that the `zerver/views/documentation.py` logic which decides whether to render the "404 error page" needs to be doing this parsing, not this code, since otherwise it's possible to reach the AssertionError in `get_endpoint_from_operationid`.

Ideally, we do it in one place, but it's not the end of the world to do it twice.  That said, if we are doing it in more than one place, we should have a function rather than copying this in N places.